recipe/meta.yaml

```diff
@@ -47,21 +47,22 @@ requirements:
     - autoconf     # [unix]
     - make         # [unix]
     - groff        # [unix]
-    - patch        # [osx]
+    - patch        # [not win]
+    - m2-patch    # [win]
     - git          # [unix]
     - m2-bash      # [win]
     - m2-patch     # [win]
     - m2-gcc-libs  # [win]
   host:
-    - krb5 1.19.*
-    - openssl 1.1.1
+    - krb5 1.20.1
+    - openssl {{ openssl }}
     - libdb ==6.2.*  # [win]
     - libntlm 1.5    # [unix]
     - sqlite 3       # [win]
     - ldap3 2.9.1    # [win]
   run:
     - krb5
```

Comment:
could you pin the runtime versions relative to the host versions? e.g. by using `pin_compatible` or pinning them with an explicit range (depending on the package's versioning scheme wrt compatibility)?

Reply:
`krb5` as a `run_exports` set to `{{ pin_subpackage('krb5', max_pin='x.x') }}`. So it will technically be pinned automatically.

But I personally don't like these implicit pins that we have all over the place. It makes it impossible to reason about our packages/recipes. Which is why I am setting `openssl 3.*` even if openssl has a run_export. I guess I could use `pin_compatible` instead? I tested it and it gives the same result as no pin at all (I guess because of the run_export).

Any opinions?

Reply:
ah right. yeah I agree about the implicit pinning.

If there's a run_export range as well, then the resulting range will be an `AND` of the run_export range and the run range in the feedstock. So it's still somewhat implicit.

Another option could be simply to repeat the run_export pinning here. Perhaps with a note saying where the pinnings come from. But it's kind of a matter of tradeoffs with no perfect option, so I'll just approve and let you do whichever you prefer